scripts/common/longhorn.sh

```diff
@@ -52,3 +52,149 @@ function longhorn_install_nfs_utils_if_missing_common() {
+            # run the migration (setting defaults)
+            $BIN_PVMIGRATE --source-sc "$longhorn_sc" --dest-sc "$destStorageClass" --rsync-image "$KURL_UTIL_IMAGE" --set-defaults
+        fi
+    done
+
+    # reset prometheus (and ekco) scale
+    if kubectl get namespace monitoring &>/dev/null; then
+        if kubectl get prometheus -n monitoring k8s &>/dev/null; then
+            if kubernetes_resource_exists kurl deployment ekc-operator; then
+                kubectl -n kurl scale deploy ekc-operator --replicas=1
+            fi
+
+            kubectl patch prometheus -n monitoring  k8s --type='json' --patch '[{"op": "replace", "path": "/spec/replicas", value: 2}]'
+        fi
+    fi
+
+    # print success message
+    printf "${GREEN}Migration from longhorn to %s completed successfully!\n${NC}" "$scProvisioner"
+    report_addon_success "longhorn-to-$scProvisioner-migration" "v1"
+}
+
+# if PVCs and object store data have both been migrated from longhorn and longhorn is no longer specified in the kURL spec, remove longhorn
+function maybe_cleanup_longhorn() {
+    if [ -z "$LONGHORN_VERSION" ]; then
+        if [ "$DID_MIGRATE_LONGHORN_PVCS" == "1" ]; then
+            report_addon_start "longhorn-removal" "v1"
+            remove_longhorn
+            report_addon_success "longhorn-removal" "v1"
+        fi
+    fi
+}
+
+# longhorn_pvs_removed returns true when we can't find any pv using the longhorn csi driver.
+function longhorn_pvs_removed() {
+    local pvs
+    pvs=$(kubectl get pv -o=jsonpath='{.items[*].spec.csi.driver}' | grep "longhorn" | wc -l)
+    [ "$pvs" = "0" ]
+}
+
+# remove_longhorn deletes everything longhorn releated: deployments, CR objects, and CRDs.
+function remove_longhorn() {
+    # make sure there aren't any PVs using longhorn before deleting it
+    echo "Waiting for Longhorn PVs to be removed"
+    if ! spinner_until 60 longhorn_pvs_removed; then
+        # sometimes longhorn hangs and we need to restart kubelet to make it work again, we
+        # are going to give this approach a try here before bailing out.
+        printf "${YELLOW}Some Longhorn PVs are still online, trying to restart kubelet.\n${NC}"
+        systemctl restart kubelet
+        echo "Waiting for Longhorn PVs to be removed"
+        if ! spinner_until 60 longhorn_pvs_removed; then
+            printf "%b" "$RED"
+            printf "ERROR: \n"
+            printf "There are still PVs using Longhorn.\n"
+            printf "Remove these PVs before continuing.\n"
+            printf "%b" "$NC"
+            exit 1
+        fi
+    fi
+
+    # scale ekco to 0 replicas if it exists
+    if kubernetes_resource_exists kurl deployment ekc-operator; then
+        kubectl -n kurl scale deploy ekc-operator --replicas=0
+        echo "Waiting for ekco pods to be removed"
+        spinner_until 120 ekco_pods_gone
+    fi
+
+    # remove longhorn volumes first so the operator can correctly delete them.
+    printf "Removing Longhorn volumes:\n"
```

Comment:
you could have used `echo` to save some keystrokes.

Reply:
could it be logStep instead ? 
```suggestion
    logStep "Removing Longhorn volumes:"
```

Reply:
fixed.